lab-raziyeh/test/book-test.js

```diff
@@ -0,0 +1,169 @@
+          expect(res.text).equal('BadRequestError');
+          expect(res.err).not.be.null;
+          done();
+        });
+      });
+    });
+  });
+
+  describe('POST requests tests', function() {
+    describe('testing POST with a valid body /api/book', function() {
+
+      after(done => {
+        if(tempBook) {
+          Book.deleteItem('book', tempBook.id)
+          .then(() => done())
+          .catch(err => done(err));
+        }
+      });
+
+      it('expecting save a new Book - POST', (done) => {
+        request.post(`${url}/api/book`)
+        .send(tempBook)
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(!!res.body.id).to.equal(true);
+          expect(res.body.title).to.equal(tempBook.title);
+          expect(res.body.author).to.equal(tempBook.author);
+          this.tempBook = res.body;
+          done();
+        });
+      });
+    });
+
+    describe('testing POST /api/book', function(){
+      it('expecting respond with bad request - POST', function(done){
+        request.post(`${url}/api/book`)
+        .send({ISBN:'445555444'})
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          expect(res.text).to.be.equal('BadRequestError');
+          done();
+        });
+      });
+    });
+  });
+
+  describe('Testing PUT requests to api/book', function(){
+  // PUT - test 200, response body like {<data>} for a post request with a valid body
+    describe('valid id and body', function() {
+      before( done => {
+        Book.createItem(exampleBook)
+        .then(book => {
+          this.tempBook = book;
+          done();
+        })
+        .catch(err => done(err));
+      });
+
+      after(done => {
+        if(this.tempBook) {
+          Book.deleteItem(this.tempBook.id)
+         .then(() => done())
+         .catch(err => done(err));
+        }
+      });
+
+      // it('should return an book update', (done) => {
+      //   let updateData = { title: 'up date', author: 'fizzbuzz', page:'222' };
+      //   request.put(`${url}/api/book?id=${this.tempBook.id}`)
+      //  .send(updateData)
+      //  .end((err, res) => {
+      //    if (err) return done(err);
+      //    expect(res.status).to.equal(200);
+      //    expect(res.body.id).to.equal(this.tempBook.id);
+      //    for (var key in updateData) {
+      //      expect(res.body[key]).to.equal(updateData[key]);
+      //    }
+      //    done();
+      //  });
+      // });
+    });
+  // PUT - test 400, responds with 'bad request' for if no body provided or invalid body
+    describe('Expecting bad request - PUT', function(){
+      it('expecting respond with bad request - PUT', function(done){
+        request.put(`${url}/api/book?id=${tempBook.id}`)
+        .send({ISBN:'445555444'})
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+  });
+});
```

Comment:
@Raziyehbazargan overall - Tests look good!  Seems like you are making a ton of progress - writing tests are hard!